csharp/ql/src/semmle/code/csharp/dataflow/LibraryTypeDataFlow.qll

```diff
@@ -500,29 +501,17 @@ private module FrameworkDataFlowAdaptor {
 }
 
 /** Data flow for `System.Int32`. */
-class SystemInt32Flow extends LibraryTypeDataFlow, SystemInt32Struct {
-  override predicate callableFlow(
-    CallableFlowSource source, CallableFlowSink sink, SourceDeclarationCallable c,
-    boolean preservesValue
-  ) {
-    methodFlow(source, sink, c) and
-    preservesValue = false
-  }
-
-  private predicate methodFlow(
-    CallableFlowSource source, CallableFlowSink sink, SourceDeclarationMethod m
-  ) {
-    m = getParseMethod() and
-    source = TCallableFlowSourceArg(0) and
-    sink = TCallableFlowSinkReturn()
-    or
-    m = getTryParseMethod() and
-    source = TCallableFlowSourceArg(0) and
-    (
-      sink = TCallableFlowSinkReturn()
-      or
-      sink = TCallableFlowSinkArg(any(int i | m.getParameter(i).isOutOrRef()))
-    )
+private class SystemInt32FlowModelCsv extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "System;Int32;false;Parse;;;Argument[0];ReturnValue;taint",
+        "System;Int32;false;TryParse;;;Argument[0];ReturnValue;taint",
```

Comment:
Should be removed.

Reply:
We did have the matching definition:
```
    m = getTryParseMethod() and
    source = TCallableFlowSourceArg(0) and
    (
      sink = TCallableFlowSinkReturn()
      or
```

Reply:
Right. But now we have the more precise summaries defined using signatures.

Reply:
Ah, sorry. I misread the `ReturnValue` as `Argument[1]`.